Heimdallr/Core/Heimdallr.swift

```diff
@@ -192,9 +195,18 @@ public let HeimdallrErrorNotAuthorized = 2
     ///
     /// **Note:** The completion closure may be invoked on any thread.
     ///
+    /// **Note:** Calls to this function are automatically serialized
+    ///
     /// - parameter request: An unauthenticated NSURLRequest.
     /// - parameter completion: A callback to invoke with the authenticated request.
     public func authenticateRequest(request: NSURLRequest, completion: Result<NSURLRequest, NSError> -> ()) {
+        dispatch_sync(refreshQueue) {
```

Comment:
Please note that if the completion block is called immediately, e.g. because there is no access token, and the completion block itself tries to authenticate another request, the program will deadlock. However, since the API is asynchronous anyways, you might want to use `dispatch_async` instead.